packages/dict/src/dict.ts

```diff
@@ -0,0 +1,54 @@
+import { Predicate, Guard, Reducer, UnaryFn } from '@proem/function'
+
+export interface Dictionary {
+  [key:string]: any
+}
+
+export const map = <A extends Dictionary, B>(dict: A, mapfn: ([k, v]: [string, any]) => B): B[] => {
+         const dictEntries = Object.entries(dict)
+         const result = new Array<B>(dictEntries.length)
+         for (let i = 0; i < dictEntries.length; i++) {
+           result[i] = mapfn(dictEntries[i])
+         }
+         return result
+       }
+
+map.partial = <A, B>(mapFn: UnaryFn<[string, any], B>) => (dict: A): B[] => map(dict, mapFn)
+
+
+export function filter<A extends Dictionary, B extends Dictionary>(dict: A, predicate: Predicate<[string, any]>): B {
```

Comment:
Same comments as with map. Generic parameter B is unnecessary. Return value should be `Dictionary<A>`